android/transit/ui/src/main/java/com/trafi/maas/transit/ui/NearbyTransitFilterItem.kt

```diff
@@ -0,0 +1,164 @@
+package com.trafi.maas.transit.ui
+
+import androidx.annotation.DrawableRes
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.sizeIn
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.trafi.core.model.ext.FilterItem
+import com.trafi.core.model.mock.mockFilters
+import com.trafi.maas.transit.ui.internal.NearbyTransitFilterItemConstants
+import com.trafi.ui.component.MultiSelectFilter
+import com.trafi.ui.component.SingleSelectFilter
+import com.trafi.ui.theme.MaasTheme
+import com.trafi.ui.theme.currentTheme
+import com.trafi.ui.theme.internal.toColor
+import com.trafi.ui.theme.isRound
+
+@Composable
+private val constants
+    get() = NearbyTransitFilterItemConstants(currentTheme)
+
+@Composable
+fun NearbyTransitFilterItem(
+    item: FilterItem,
+    enabled: Boolean,
+    modifier: Modifier = Modifier,
+) {
+    Box(
+        contentAlignment = Alignment.Center,
+        modifier = modifier.sizeIn(
+            minWidth = constants.itemMinWidth,
+            minHeight = constants.itemMinHeight
+        )
+    ) {
+        Surface(
+            shape = if (constants.cornerRadius.isRound) {
+                RoundedCornerShape(percent = 50)
+            } else {
+                RoundedCornerShape(constants.cornerRadius)
+            },
+            color = if (enabled) item.color.toColor() else constants.disabledColor,
+            modifier = Modifier
+                .sizeIn(
+                    minWidth = constants.contentMinWidth,
+                    minHeight = constants.contentMinHeight
+                )
+        ) {
+            val image = vectorResource(id = item.icon.iconRes())
+            Image(
+                imageVector = image,
+                modifier = Modifier
+                    .width(constants.imageWidth)
+                    .height(constants.imageHeight),
+                colorFilter = ColorFilter.tint(color = item.accentColor.toColor())
+            )
+        }
+    }
+}
+
+@DrawableRes
+private fun String.iconRes(): Int {
```

Comment:
@justasm Maybe there is a possibility to move these mappings in some general module as ui or other to make it reachable in all modules because the same mapping is used in RoutesSegment composable. I guess there will be more similar functions in the future. What do you think?

Reply:
Yeah, I agree. We can move it around and consolidate it but it doesn't solve the issue that we're bundling Berlin-specific transport icons. We'll address this in a more holistic way when we tackle image / icon loading in the SDK.

Reply:
The mapping in `RouteSegment` maps from `Schedule.transportType` to icon; here we map from `FilterItem.icon`. It's a coincidence that the strings match 🙂 It's not clear if we want to load the same assets for route segment badges as we do for filter items.

Reply:
Good point 👍